controller/apps.go

```diff
@@ -0,0 +1,413 @@
+	data := make([]*app.TimedIntTuple, limit, limit)
+
+	for i := 0; i < limit; i++ {
+		t := start + int(period*float64(i))
+		v := int(float64(high-low) * float64(i) / float64(limit))
+
+		tuple := app.TimedIntTuple{
+			Time:  &t,
+			Value: &v,
+		}
+		data[i] = &tuple
+	}
+	return data
+}
+
+// ShowDeploymentStatSeries runs the showDeploymentStatSeries action.
+func (c *AppsController) ShowDeploymentStatSeries(ctx *app.ShowDeploymentStatSeriesAppsContext) error {
+	// AppsController_ShowDeploymentStatSeries: start_implement
+
+	endMillis := time.Now().UnixNano() / 1000000
+	var eightHoursMillis int64 = 8 * 60 * 60 * 1000
+	startMillis := endMillis - eightHoursMillis
+	limit := 10
+
+	if ctx.Limit != nil {
+		limit = *ctx.Limit
+	}
+
+	if ctx.Start != nil {
+		startMillis = int64(*ctx.Start)
+	}
+
+	if ctx.End != nil {
+		endMillis = int64(*ctx.End)
+	}
+
+	if endMillis < startMillis {
+		return errors.NewBadParameterError("end", *ctx.End)
+	}
+
+	startInt := int(startMillis)
+	endInt := int(endMillis)
+	cores := genData(startInt, endInt, limit, 0, 10)
+	memory := genData(startInt, endInt, limit, 1000000, 2000000)
+	res := &app.SimpleDeploymentStatSeries{
+		Start:  &startInt,
+		End:    &endInt,
+		Cores:  cores,
+		Memory: memory,
+	}
+
+	// AppsController_ShowDeploymentStatSeries: end_implement
+	return ctx.OK(res)
+}
+
+// ShowDeploymentStats runs the showDeploymentStats action.
+func (c *AppsController) ShowDeploymentStats(ctx *app.ShowDeploymentStatsAppsContext) error {
+	// AppsController_ShowDeploymentStats: start_implement
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	kubeSpaceName, err := c.getSpaceNameFromSpaceID(ctx, ctx.SpaceID)
+	if err != nil {
+		return err
+	}
+
+	deploymentStats, err := kc.GetDeployment(*kubeSpaceName, ctx.AppName, ctx.DeployName)
+	if err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	if deploymentStats == nil {
+		return errors.NewNotFoundError("deployment", ctx.DeployName)
+	}
+
+	res := &app.SimpleDeploymentSingle{
+		Data: deploymentStats,
+	}
+
+	// AppsController_ShowDeploymentStats: end_implement
+	return ctx.OK(res)
+}
+
+// ShowEnvironment runs the showEnvironment action.
+func (c *AppsController) ShowEnvironment(ctx *app.ShowEnvironmentAppsContext) error {
+	// AppsController_ShowEnvironment: start_implement
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	env, err := kc.GetEnvironment(ctx.EnvName)
+	if err != nil {
+		return errors.NewInternalError(ctx, err)
+	}
+	if env == nil {
+		return errors.NewNotFoundError("environment", ctx.EnvName)
+	}
+
+	res := &app.SimpleEnvironmentSingle{
+		Data: env,
+	}
+
+	// AppsController_ShowEnvironment: end_implement
+	return ctx.OK(res)
+}
+
+// ShowSpace runs the showSpace action.
+func (c *AppsController) ShowSpace(ctx *app.ShowSpaceAppsContext) error {
+	// AppsController_ShowSpace: start_implement
+
+	kc, err := c.getAndCheckKubeClient(ctx)
+	if err != nil {
+		return err
```

Comment:
same comment about error conversion

Reply:
True - but please see my reply to your comment on getAndCheckKubeClient()